src/extension/tools/node/findTextInFilesTool.tsx

```diff
@@ -72,17 +77,35 @@ export class FindTextInFilesTool implements ICopilotTool<IFindTextInFilesToolPar
 		// If we still have no results, we need to try the opposite regex mode
 		if (!results.length && queryIsValidRegex) {
 			results = await raceTimeoutAndCancellationError(
-				(searchToken) => this.searchAndCollectResults(options.input.query, !isRegExp, patterns, maxResults, searchToken),
+				(searchToken) => this.searchAndCollectResults(options.input.query, !isRegExp, patterns, maxResults, includeIgnoredFiles, searchToken),
 				token,
 				timeoutInMs,
 				// embed message to give LLM hint about what to do next
 				`Find ${results.length} results in searching text in files with ${isRegExp ? 'regex' : 'literal'} search, and then another searching hits timeout in with ${!isRegExp ? 'regex' : 'literal'} search, try a more specific search pattern`
 			);
 		}
 
+		let noMatchInstructions: string | undefined = undefined;
+		if (!results.length && !includeIgnoredFiles) {
+			// Get the search.exclude configuration
+			const excludeSettings = this.configurationService.getNonExtensionConfig<Record<string, boolean>>('search.exclude');
+			const excludePaths: string[] = [];
+			if (excludeSettings) {
+				for (const [path, isExcluded] of Object.entries(excludeSettings)) {
+					if (isExcluded) {
+						excludePaths.push(path);
+					}
+				}
+			}
+
+			noMatchInstructions = `Your search pattern might be excluded completedly by either the search.exclude settings or .*ignore files.
+			If you believe that it should have results, you can check into the .*ignore files and the exclude setting (here are some excluded patterns for reference:[${excludePaths.join(';')}] separated by ';').
```

Comment:
Does this seem necessary, or can it figure out how to fall back to use this parameter without the extra hint?

Reply:
The extra hint will make it more stable from my tests. Let me try shorten it by using comma.